src/Microsoft.Health.Fhir.Azure/ExportDestinationClient/AzureAccessTokenClientInitializer.cs

```diff
@@ -7,42 +7,37 @@
 using System.Net;
 using System.Threading;
 using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Storage.Blobs;
 using EnsureThat;
-using Microsoft.Azure.Storage.Auth;
-using Microsoft.Azure.Storage.Blob;
-using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using Microsoft.Health.Fhir.Core.Configs;
 using Microsoft.Health.Fhir.Core.Features.Operations.Export.ExportDestinationClient;
 
 namespace Microsoft.Health.Fhir.Azure.ExportDestinationClient
 {
-    public class AzureAccessTokenClientInitializer : IExportClientInitializer<CloudBlobClient>
+    public class AzureAccessTokenClientInitializer : IExportClientInitializer<BlobServiceClient>
     {
-        private readonly IAccessTokenProvider _accessTokenProvider;
+        private readonly TokenCredential _credential;
         private readonly ExportJobConfiguration _exportJobConfiguration;
-        private readonly ILogger<AzureAccessTokenClientInitializer> _logger;
 
         public AzureAccessTokenClientInitializer(
-            IAccessTokenProvider accessTokenProvider,
-            IOptions<ExportJobConfiguration> exportJobConfiguration,
-            ILogger<AzureAccessTokenClientInitializer> logger)
+            TokenCredential credential,
+            IOptions<ExportJobConfiguration> exportJobConfiguration)
         {
-            EnsureArg.IsNotNull(accessTokenProvider, nameof(accessTokenProvider));
+            EnsureArg.IsNotNull(credential, nameof(credential));
             EnsureArg.IsNotNull(exportJobConfiguration?.Value, nameof(exportJobConfiguration));
-            EnsureArg.IsNotNull(logger, nameof(logger));
 
-            _accessTokenProvider = accessTokenProvider;
```

Comment:
Is there a specific reason we do not want to use this interface anymore. The idea behind it was to allow an easy way to change the logic of acquiring access tokens by injecting a different implementation of `IAccessTokenProvider`.

Reply:
AFAIK, all the azure libraries are now using `Azure.Core` for auth, hence `TokenCredential` is accepted as a parameter. I didn't see an easy way to use `IAccessTokenProvider` here.
Also `TokenCredential` is an abstract class, so in the sense of injecting different implementation, you can still do that.

Reply:
Actually, there is even a diclaimer that it should not be used anymore: https://docs.microsoft.com/en-us/dotnet/api/overview/azure/service-to-service-authentication